app/models/measure.rb

```diff
@@ -61,10 +61,20 @@ def queue_task_updated_notifications!(user_id: ::PaperTrail.request.whodunnit)
     delete_existing_task_notifications!(user_id:)
 
     notifiable_user_measures(user_id:).each do |user_measure|
-      TaskNotificationJob.perform_in(ENV.fetch("TASK_NOTIFICATION_DELAY", 20).to_i.seconds, user_measure.id)
+      queue_task_updated_notification!(user_id: user_measure.user_id, measure_id: user_measure.measure_id)
     end
   end
 
+  def queue_task_updated_notification!(user_id:, measure_id:)
+    return unless notify?
+
```

Comment:
should we also the `delete_existing_task_notifications` here?

Reply:
have updated that now

Reply:
I've updated that, and made the outer `each` loop instruct the method not to as it's already handled by `queue_task_updated_notifications!` and otherwise it'll delete existing notifications for each iteration of the loop, which I think we don't want?

Reply:
@tmfrnz I've updated that, and made the outer `each` loop instruct the method not to as it's already handled by `queue_task_updated_notifications!` and otherwise it'll delete existing notifications for each iteration of the loop, which I think we don't want?

Reply:
@tmfrnz what I mean is that when this is called from `queue_task_updated_notifications!` which also calls `delete_existing_task_notifications` we don't want to call it again here because then it would only create a notification for the last measure in the list, because all of the other ones will get deleted.

Let me know if that doesn't make sense. 😁

Reply:
sounds good - thank you @parndt for explaining!